src/main/scala/mesosphere/marathon/api/v2/AppTasksResource.scala

```diff
@@ -146,20 +159,22 @@ class AppTasksResource @Inject() (
       val deploymentF = taskKiller.killAndScale(pathId, findToKill, force)
       deploymentResult(result(deploymentF))
     } else {
-      reqToResponse(taskKiller.kill(pathId, findToKill, wipe)) {
-        tasks => tasks.headOption.fold(unknownTask(id))(task => ok(jsonObjString("task" -> task)))
+      val response: Future[Response] = async {
+        val instances = await(taskKiller.kill(pathId, findToKill, wipe))
+        val healthStatuses = await(healthCheckManager.statuses(pathId))
+        instances.headOption match {
+          case None =>
+            unknownTask(id)
+          case Some(instance) =>
+            val killedTask = instance.appTask // Should this rather be instance.taskMap[taskId]?
```

Comment:
The `appTask` helper was introduced to prevent accessing the map (and helps identifying the locations where we do that). I'd either stick to the helper or get rid of it; a mixture of both is unfortunate.

Reply:
Ok, I stick with the old.